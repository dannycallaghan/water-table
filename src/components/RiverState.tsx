function RiverState(props: { state: string }) {
  let classes =
    'flex items-center justify-center rounded bg-green-400 px-1 text-[10px] uppercase text-white';

  if (props.state.toLowerCase() === 'rising') {
    classes =
      'flex items-center justify-center rounded bg-yellow-400 px-1 text-[10px] uppercase text-slate-800';
  }

  if (props.state.toLowerCase() === 'falling') {
    classes =
      'flex items-center justify-center rounded bg-red-400 px-1 text-[10px] uppercase text-white';
  }

  return <div className={classes}>{props.state}</div>;
}

export default RiverState;
