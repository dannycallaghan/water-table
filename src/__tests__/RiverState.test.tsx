import { expect, test, describe } from 'vitest';
import { render } from '@testing-library/react';
import RiverState from '../components/RiverState.tsx';

describe('RiverState', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<RiverState state={'Falling'} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders correctly', () => {
    const { asFragment } = render(<RiverState state={'Constant'} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders correctly', () => {
    const { asFragment } = render(<RiverState state={'Rising'} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
