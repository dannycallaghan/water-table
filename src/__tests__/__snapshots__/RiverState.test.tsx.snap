// Vitest Snapshot v1

exports[`RiverState > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center rounded bg-red-400 px-1 text-[10px] uppercase text-white"
  >
    Falling
  </div>
</DocumentFragment>
`;

exports[`RiverState > renders correctly 2`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center rounded bg-green-400 px-1 text-[10px] uppercase text-white"
  >
    Constant
  </div>
</DocumentFragment>
`;

exports[`RiverState > renders correctly 3`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center rounded bg-yellow-400 px-1 text-[10px] uppercase text-slate-800"
  >
    Rising
  </div>
</DocumentFragment>
`;
